Geometry/HcalCommonData/BuildFile.xml

```diff
@@ -1,6 +1,7 @@
 <use   name="DetectorDescription/Base"/>
 <use   name="DetectorDescription/Core"/>
 <use   name="DataFormats/HcalDetId"/>
+<use   name="SimDataFormats/CaloTest"/>
```

Comment:
@ianna I previously had the dependency the other way around, but @bsunanda suggested that DataFormats shouldn't depend on Geometry. There's no `PCaloHit` dependence in `HcalHitRelabeller`. Maybe the HcalTestNumbering class should also be moved to Geometry/HcalCommonData? Then we might have to move `HcalTestHistoClass` also... @bsunanda, @abdoulline, is that class still used?